test/src/tests-consent.ts

```diff
@@ -0,0 +1,750 @@
+                'foo hardware id'
+            );
+        expect(consent).to.be.ok;
+        consent.should.have.property('Consented', true);
+        consent.should.have.property('Timestamp', 10);
+        consent.should.have.property('ConsentDocument', 'foo document');
+        consent.should.have.property('Location', 'foo location');
+        consent.should.have.property('HardwareId', 'foo hardware id');
+        done();
+    });
+    // to here
+    it('Should create basic ConsentState object', done => {
+        const consentState = mParticle
+            .getInstance()
+            .Consent.createConsentState();
+        expect(consentState).to.be.ok;
+        done();
+    });
+
+    it('Should add GDPR ConsentState object', done => {
+        const consentState = mParticle
+            .getInstance()
+            .Consent.createConsentState();
+        expect(consentState).to.be.ok;
+
+        consentState
+            .addGDPRConsentState('foo', (mParticle.Consent.createGDPRConsent(
+                true
+            ) as unknown) as PrivacyConsentState)
+            .addGDPRConsentState('bar', (mParticle.Consent.createGDPRConsent(
+                false
+            ) as unknown) as PrivacyConsentState);
+
+        expect(consentState.getGDPRConsentState()).to.have.property('foo');
+
+        // FIXME: getGDPRConsentState should return a dictionary of consent states
+        //        rather than a consent state
+        expect(
+            ((consentState.getGDPRConsentState() as unknown) as GDPRConsentStateDictionary)
+                .foo
+        ).to.have.property('Consented', true);
+
+        // Test is verifying that attribute GDPR does not exist in ConsentState
+        // but this makes TypeScript mad.
+        expect((consentState as any).GDPR === undefined).to.be.ok;
+
+        expect(consentState.getGDPRConsentState()).to.have.property('bar');
+        expect(
+            ((consentState.getGDPRConsentState() as unknown) as GDPRConsentStateDictionary)
+                .bar
+        ).to.have.property('Consented', false);
+
+        done();
+    });
+
+    it('Should replace GDPR ConsentState object', done => {
+        const consentState = mParticle
+            .getInstance()
+            .Consent.createConsentState();
+        const consentState2 = mParticle
+            .getInstance()
+            .Consent.createConsentState();
+
+        consentState.addGDPRConsentState(
+            'foo',
+            (mParticle.Consent.createGDPRConsent(
+                true
+            ) as unknown) as PrivacyConsentState
+        );
+
+        consentState2.addGDPRConsentState(
+            'bar',
+            (mParticle.Consent.createGDPRConsent(
+                false
+            ) as unknown) as PrivacyConsentState
+        );
+        consentState2.addGDPRConsentState(
+            'baz',
+            (mParticle.Consent.createGDPRConsent(
+                false
+            ) as unknown) as PrivacyConsentState
+        );
```

Comment:
The need for this workaround is because I'm splitting updating tests from updating the actual core Consent module. We can move these to the top of the tests with a single TODO that is tied to that ticket.

Reply:
After reviewing the Consent API, the reason I'm casting these as `PrivacyConsentState` is because `addGDPRConsentState` doesn't recognize the return of `createGDPRConsent` (which is `GDPRConsentState`) is a type of `PrivacyConsentState`. May need to revisit this.